javagen/src/main/java/com/azure/autorest/template/ConvenienceAsyncMethodTemplate.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.autorest.template;
+
+import com.azure.autorest.model.clientmodel.ClassType;
+import com.azure.autorest.model.clientmodel.ClientMethod;
+import com.azure.autorest.model.clientmodel.ConvenienceMethod;
+import com.azure.autorest.model.clientmodel.EnumType;
+import com.azure.autorest.model.clientmodel.GenericType;
+import com.azure.autorest.model.clientmodel.IType;
+import com.azure.autorest.model.javamodel.JavaBlock;
+import com.azure.autorest.util.ClientModelUtil;
+
+public class ConvenienceAsyncMethodTemplate extends ConvenienceMethodTemplateBase {
+
+    private static final ConvenienceAsyncMethodTemplate INSTANCE = new ConvenienceAsyncMethodTemplate();
+
+    protected ConvenienceAsyncMethodTemplate() {
+    }
+
+    public static ConvenienceAsyncMethodTemplate getInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    protected boolean isConvenienceMethod(ClientMethod method) {
+        return isMethodAsync(method) && isMethodVisible(method);
+    }
+
+    @Override
+    protected boolean isConvenienceMethod(ConvenienceMethod method) {
+        return isMethodAsync(method.getClientMethod()) && isMethodVisible(method.getClientMethod())
+                && method.getClientMethod().getMethodParameters().stream().noneMatch(p -> p.getClientType() == ClassType.Context);
```

Comment:
Why do we add `Context` for async client method, but not add for sync client method?

Reply:
Async have `Context` in mono/flux.